service/collector/helper.go

```diff
@@ -183,3 +181,36 @@ func (h *helper) AWSAccountID(awsClients clientaws.Clients) (string, error) {
 
 	return accountID, nil
 }
+
+// ListReconciledClusters provides names of clusters reconciled by this
+// particular operator version. The names are returned in format of a map,
+// instead of a list, for cheaper lookup.
+func (h *helper) ListReconciledClusters() (reconciled map[string]bool) {
```

Comment:
I am fine with the helper but this is somehow not aligned with what I thought we discussed. This should return the CRs. I would like to have something like this. 


```
func (a *ASG) Collect(ch chan<- prometheus.Metric) error {
    list, err := a.helper.ListReconciledClusters()

    // TODO get all ASGs of the installation.

    for _, cr := range list.Items {
        // TODO create AWS clients for Tenant Cluster. 

        // TODO emit metrics
    }

    return nil
}
```

This design has the benefits of the code being more straight forward and us doing less calls to AWS. Let's maybe pair a bit if we do not agree on this. I do not want to make your life harder. ❤️

Reply:
Yeah, it's not me disagreeing with you. It's me not understanding the system 100%. Now I get it - we used to collect ARNs from *all* the clusters and build client list off of that.

I've pushed new commits, which move this part around. I considered packing the `ListReconciledClusters` call to `GetARN` directly, but then we would not be able to query all the resources for ELBs for example.

That's why I'm passing it all the way down right now.

Reply:
So do we want to change the AWS API calls here as well or not? I am fine with doing things separately and I think there is more refactoring to it anyway. Let's just get it straight so we are on the same page.

Reply:
Discussed outside of Github, decided on limiting scope to reconciled cluster filtering.